src/spec-configuration/containerFeaturesOrder.ts

```diff
@@ -31,7 +31,7 @@ export function computeOverrideInstallationOrder(config: DevContainerConfig, fea
     // Moves to the beginning the features that are explicitly configured.
     const orderedFeatures = [];
     for (const featureId of config.overrideFeatureInstallOrder!) {
-        const feature = automaticOrder.find(feature => feature.features[0].name === featureId);
+        const feature = automaticOrder.find(feature => (feature.sourceInformation.type === 'oci' && feature.sourceInformation.userFeatureId.split(':')[0] === featureId) || (feature.sourceInformation.type === 'github-repo' && feature.sourceInformation.userFeatureId.split('@')[0] === featureId) || feature.sourceInformation.userFeatureId === featureId);
```

Comment:
Remind me why this needs to change, and why we're filtering for only OCI features?

Reply:
We are changing this to match with the [spec](https://github.com/devcontainers/spec/blob/main/proposals/devcontainer-features.md#1-overridefeatureinstallorder). (ref issue - https://github.com/devcontainers/images/pull/53#discussion_r939429374 )

> why we're filtering for only OCI features
 We are filtering for all the types, `userFeatureId` follows this [part of spec](https://github.com/devcontainers/spec/blob/main/proposals/devcontainer-features.md#referencing-a-feature), however for `OCI` the `overridefeatureinstallorder` needs unversioned feature IDs, hence the special condition.

Reply:
I would prefer breaking up this long condition into a couple local variables for readability.  

Better yet, break up the individual components (location bit and version bit) and store them on the SourceInformation as well. That will make it easier to unit test.

Reply:
> Better yet, break up the individual components (location bit and version bit) and store them on the SourceInformation as well.

Done!